### PR TITLE
Updates to BMA functionality 

### PR DESCRIPTION
**Pull request by drnickisaac**

### CODE REVIEW

**Comment by AugustT** [comment]
Fixed a conflict in the description file (we both updated the authors)

**Comment by AugustT** [comment]
@drnickisaac the example in bma fails

```r
data <- data.frame(species = rep(letters, each = 50),
                    year = rep(1:50, length(letters)),
                    index = runif(n = 50 * length(letters), min = 0, max = 1),
                    se = runif(n = 50 * length(letters), min = 0.01, max = .1))

 # Run the Bayesian meta-analysis
 bma_indicator <- bma(data, model="Smooth")
```
```
Error in tolower(model) : object 'model' not found
```

File: R/get_bmaBUGScode.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#' @param option text string defining which particular variant is desired 
+#' @export
+#' @examples
+#' extract_bmaBUGScode(option="Smooth", print.screen=TRUE, save.local=FALSE)
+
+get_bmaBUGScode <- function(option="Smooth", 
+                                print.screen = FALSE,
+                                save.local = FALSE) {
+  
+  switch(tolower(model),
```
Comment (AugustT) [inline R/get_bmaBUGScode.R]: @drnickisaac This function would never run without error. object `model` does not exist, you mean `option`. Make sure you run your examples to ensure they work.

Comment (AugustT) [reply R/get_bmaBUGScode.R]: I have implemented the fix

File: R/get_bmaBUGScode.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#' @export
+#' @examples
+#' extract_bmaBUGScode(option="Smooth", print.screen=TRUE, save.local=FALSE)
+
+get_bmaBUGScode <- function(option="Smooth", 
+                                print.screen = FALSE,
+                                save.local = FALSE) {
+  
+  switch(tolower(option),
+         Smooth = {model_code <- bma_model_Smooth()},
```
Comment (AugustT) [inline R/get_bmaBUGScode.R]: @drnickisaac you tolower so you dont want a capital

File: R/get_bmaBUGScode.R [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+         uniform_noeta = {model_code <- bma_model_uniform_noeta()},
+         
+         fngr = {model_code <- bma_model_FNgr()},
+         smooth_stoch = {model_code <- bma_model_smooth_stoch()},
+         smooth_det = {model_code <- bma_model_smooth_det()},
+         fngr2 = {model_code <- bma_model_FNgr2()},
+         smooth_stoch2 = {model_code <- bma_model_smooth_stoch2()},
+         smooth_det2 = {model_code <- bma_model_smooth_det2()},
+         smooth_det_sigtheta = {model_code <- bma_model_smooth_det_sigtheta()},
+         {stop(paste("model type not know. Must be one of 'random_walk',",
```
Comment (AugustT) [inline R/get_bmaBUGScode.R]: @drnickisaac these dont match the options

File: R/get_bmaBUGScode.R (hunk 2 above)
Comment (AugustT) [reply R/get_bmaBUGScode.R]: implemented fix

File: R/get_bmaBUGScode.R (hunk 3 above)
Comment (AugustT) [reply R/get_bmaBUGScode.R]: fixed

**Comment by AugustT** [comment]
@drnickisaac There are more issues. Rather than me trying to tackle this can I ask you to run your examples and address all the errors that you get and then put in a new pull request?

**Comment by AugustT** [comment]
```r
# Create some example data in the format required
 data <- data.frame(species = rep(letters, each = 50),
                    year = rep(1:50, length(letters)),
                    index = runif(n = 50 * length(letters), min = 0, max = 1),
                    se = runif(n = 50 * length(letters), min = 0.01, max = .1))
 
 # Run the Bayesian meta-analysis
 bma_indicator <- bma(data, model="Smooth")

```
```
Error in jags.model(file = model.file, data = data, inits = inits, n.chains = n.chains,  : 
  
Error parsing model file:
syntax error on line 48 near "{"
```

**Comment by AugustT** [comment]
@drnickisaac nudge

**Comment by drnickisaac** [comment]
> @drnickisaac nudge

Done!

**Comment by AugustT** [comment]
Hi @drnickisaac are you not getting these errors on your local version?

```r
 data <- data.frame(species = rep(letters, each = 50),
                   year = rep(1:50, length(letters)),
                   index = runif(n = 50 * length(letters), min = 0, max = 1),
                   se = runif(n = 50 * length(letters), min = 0.01, max = .1))

# Run the Bayesian meta-analysis
bma_indicator <- bma(data, model="Smooth")
```
```
[1] "Warning: No negative index values detected. Are you sure you transformed the data?"
Processing function input....... 
Done. 

Error in jags.model(file = model.file, data = data, inits = inits, n.chains = n.chains,  : 
  
Error parsing model file:
syntax error on line 48 near "{"
```

**Comment by drnickisaac** [comment]
Doh!

1. The example data had not been converted to the logit (or log) scale. I've now changed that in the example. We should think about whether there's a better way to explain this, since the other methods take untransformed data. For now though, let's just get it working.
2. Second problem fixed. This model is perhaps not the best choice for a default, since it's the least tested! But that's a problem for next week when I start working on Steve's manuscript again.

**Comment by AugustT** [comment]
The problem persists. I suggest looking closely at your syntax once you have parsed your bugs code for `smooth`, it looks like there is a bug in there. It might be a good idea to write some tests for this function so that you can catch this before the pull request stage.

**Comment by drnickisaac** [comment]
@AugustT I think the check failure is only on code coverage, not a bug in my code. Correct?

**Comment by AugustT** [comment]
Correct, all green on Travis

**Comment by AugustT** [comment]
In she goes. Thanks for getting this over the line

**Pull request merged**